### DISCUSSION TITLE
How does PgSync guarantee not lost data ?

### DISCUSSION THREAD

**Discussion by TheNhatAT** [toluaina/pgsync#577] [Q&A]
When PG_NOTIFY happen but PgSync service is downed, so there are not any LISTEN active to listen the event, then I consider that the event will be lost.

**Comment by aadrian** [comment]
@TheNhatAT 

> so there are not any LISTEN active to listen the event, then I consider that the event will be lost.

https://www.postgresql.org/docs/current/sql-notify.html  - it will be thrown away by PG, so I guess noting a tool that does not run for a  short time can do.

Maybe with an extra "catch all log" table to write the events to it in case for whatever reason there's zero tool listener ?
